File: server/notification-service/src/main/java/com/sudoo/notification/service/DiscoveryService.java
package com.sudoo.notification.service;

import com.sudo248.domain.base.BaseResponse;
import com.sudoo.notification.controller.dto.SupplierInfoDto;
import org.springframework.cloud.openfeign.FeignClient;
import org.springframework.http.ResponseEntity;
import org.springframework.stereotype.Service;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;

@FeignClient(value = "PRODUCT-SERVICE")
@Service
public interface DiscoveryService {
    @GetMapping("/api/v1/discovery/supplier/info/{supplierId}")
    ResponseEntity<BaseResponse<SupplierInfoDto>> getSupplierInfoById(@PathVariable("supplierId") String supplierId);
}

File: server/cart-service/src/main/kotlin/com/sudoo/cartservice/internal/DiscoveryService.kt
package com.sudoo.cartservice.internal

import com.sudoo.cartservice.controller.dto.SupplierProductDto
import com.sudoo.domain.common.Constants
import org.springframework.stereotype.Service
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.RequestHeader
import org.springframework.web.bind.annotation.RequestParam

@Service
 interface DiscoveryService {
    @GetMapping("/api/v1/discovery/service/supplierId/{supplierId}/productId/{productId}")
    fun getSupplierProduct(
            @RequestHeader(Constants.HEADER_USER_ID) userId: String,
            @PathVariable("supplierId") supplierId: String,
            @PathVariable("productId") productId: String,
            @RequestParam("hasRoute") hasRoute: Boolean
    ): SupplierProductDto

    @GetMapping("/api/v1/discovery/service/supplierId/{supplierId}/productId/{productId}/price")
    fun getSupplierProductPrice(
            @PathVariable("supplierId") supplierId: String,
            @PathVariable("productId") productId: String
    ): Double
}